[QUESTION] How to use screen recording/streaming apps?

I found on https://github.com/WayfireWM/wayfire/issues/1012 that https://github.com/emersion/xdg-desktop-portal-wlr should be used for screensharing and video recording. How is this supposed to be started when using wayfire?
* I installed https://github.com/emersion/xdg-desktop-portal-wlr following their instructions, and tried the command they suggest, but nothing is displayed on-screen and there are a bunch of errors / warnings in the console (see below).
* Trying with `export XDG_CURRENT_DESKTOP=wayfire` gives the exact same output modulo the timestamps and PIDs.
* I also tried to install https://github.com/ammen99/wf-recorder, it allowed me to select the output to be recorded (I have two monitors), but it crashed the session right after pressing Enter ([wayfire.log](https://github.com/WayfireWM/wayfire/files/6098144/wayfire.log)).
* I tried installing `obs` from source, from Nix, via `flatpak`, via `apt`, and installing the `wlrobs` plug-in, none of those offer wayland as an input and half of them don't even start. I managed to capture a single window by running both `obs` and that window within the same Xwayland; it seems impossible to capture the entire Xwayland display however.

Is this:
* a `wayfire` problem ?
* a `xdg-desktop-portal-wlr` problem ?
* a `pipewire` problem ?
* something magical that must be done with `systemd` (I saw a few .service files were installed, but `sudo service xdg-desktop-portal-wlr start` says that the service does not exist, maybe a /usr vs. /usr/local problem)?
* a `wf-recorder` problem ?
* an `obs` problem?

I have tried pretty much everything I could think of, and I'm still unsure about which component is causing issues.

If there's a distro on which wayfire + some streaming app are supposed to work relatively painlessly, I'm quite willing to give it a go :)

```
export XDG_CURRENT_DESKTOP=sway; /usr/libexec/xdg-desktop-portal -r & /usr/local/libexec/xdg-desktop-portal-wlr
[1] 229889
[E][pipewire.c:335 add_interface()] can't get Spa:Pointer:Interface:Log interface -2
[E][pipewire.c:335 add_interface()] can't get Spa:Pointer:Interface:CPU interface -2
[E][loop.c:85 pw_loop_new()] loop 0x56212273e6d0: can't make support.system handle: No such file or directory
2021/03/07 22:57:09 [ERROR] - pipewire: failed to create loop

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:09.701: Failed to create settings proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.kde: Process org.freedesktop.impl.portal.desktop.kde received signal 6

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:09.859: Failed to create file chooser proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.kde: Process org.freedesktop.impl.portal.desktop.kde received signal 6

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:09.859: No skeleton to export

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:10.018: Failed to create app chooser proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.kde: Process org.freedesktop.impl.portal.desktop.kde received signal 6

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:10.018: No skeleton to export

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:10.179: Failed to create print proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.kde: Process org.freedesktop.impl.portal.desktop.kde received signal 6

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:10.179: No skeleton to export

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:10.335: Failed to create screenshot proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.kde: Process org.freedesktop.impl.portal.desktop.kde received signal 6

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:10.335: No skeleton to export

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:10.495: Failed to create notification proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.kde: Process org.freedesktop.impl.portal.desktop.kde received signal 6

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:10.495: No skeleton to export

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:10.655: Failed to create inhibit proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.kde: Process org.freedesktop.impl.portal.desktop.kde received signal 6

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:10.655: No skeleton to export

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:10.811: Failed to create access proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.kde: Process org.freedesktop.impl.portal.desktop.kde received signal 6

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:10.811: No skeleton to export

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:10.970: Failed to create access proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.kde: Process org.freedesktop.impl.portal.desktop.kde received signal 6

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:10.970: No skeleton to export

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:11.130: Failed to create email proxy: Error calling StartServiceByName for org.freedesktop.impl.portal.desktop.kde: Process org.freedesktop.impl.portal.desktop.kde received signal 6

** (/usr/libexec/xdg-desktop-portal:229889): WARNING **: 22:57:11.130: No skeleton to export
```

--- a/src/output/render-manager.cpp
+++ b/src/output/render-manager.cpp
@@ -984,9 +984,11 @@ class wf::render_manager::impl
             swap_damage |= output_damage->get_wlr_damage_box();
         }
 
-        OpenGL::render_begin(postprocessing->get_target_framebuffer());
-        wlr_output_render_software_cursors(output->handle, swap_damage.to_pixman());
-        OpenGL::render_end();
+        wlr_renderer_begin(wf::get_core().renderer,
+            output->handle->width, output->handle->height);
+        wlr_output_render_software_cursors(output->handle,
+            swap_damage.to_pixman());
+        wlr_renderer_end(wf::get_core().renderer);
 
         /* Part 4: postprocessing effects */
         postprocessing->run_post_effects();

